src/spikeinterface/postprocessing/spike_amplitudes.py

```diff
@@ -127,13 +127,20 @@ def _get_data(self, outputs="numpy"):
         elif outputs == "by_unit":
             unit_ids = self.sorting_analyzer.unit_ids
             spike_vector = self.sorting_analyzer.sorting.to_spike_vector(concatenated=False)
+            segment_cum_count = np.insert(
+                np.cumsum([len(spike_vector[a]) for a in range(0, self.sorting_analyzer.sorting.get_num_segments())]),
+                0,
+                0,
+            )
```

Comment:
@chrishalcrow I think this should be handled in the `spike_vector_to_indices`, so it's used by all extensions that do this `by_unit`

Reply:
@samuelgarcia will fix it because it's his fault!